### PR TITLE
Makefile: add types module testing

### PR DESCRIPTION
**Pull request by NDStrahilevitz**
Solves #1509

This PR:
1. Adds a script(`test-types`) to test the `types` module since it was missing from tests.
2. Adds `test-types` to `make test`.
3. Adds a job to verify types in the PR github workflow.

### CODE REVIEW

File: Makefile
Comment (mtcherni95) [inline Makefile:629]: you need to update also the `help`

**Review by mtcherni95 — CHANGES_REQUESTED** [review]

File: Makefile
Comment (mtcherni95) [inline Makefile:634]:
```suggestion
	cd ./types; $(CMD_GO) test -v ./...
```

Comment (NDStrahilevitz) [reply Makefile:634]: Thanks I couldn't find a script like that which worked haha

File: Makefile
Comment (NDStrahilevitz) [reply Makefile:630]: done

File: Makefile
Comment (mtcherni95) [inline Makefile:633]: empty line

Comment (NDStrahilevitz) [reply Makefile:633]: That empty line with # is there in almost all other scripts.

Comment (rafaeldtinoco) [reply Makefile:633]: Yep, this is following the format I proposed to Makefiles. I'll review this tomorrow morning.

Comment (mtcherni95) [reply Makefile:633]: ohh didn't know that, my bad :) thanks @rafaeldtinoco

File: Makefile
Comment (rafaeldtinoco) [inline Makefile:627]: Despite this change being functional and working good, I have concerns on "test-types". What does it mean for the end user ? What is "types" ?  We had, so far, a clear distinction between "tracee-ebpf", "tracee-rules" and "rules", thus the 3 different tests. 

Shouldn't we add the types test, for now, to "test-tracee-ebpf" OR "test-tracee-rules" ? And, after we're done with making "tracee-ebpf + tracee-rules" a single program, then we can have all tests in a single "test target".

WDYT ?

Comment (NDStrahilevitz) [reply Makefile:627]: I'm not sure it fits either programs, `types` defines their protocol. We could split the inner packages between testing ebpf and rules but that seems ugly.

I agree we can find a better solution once we merge the two into a single program, but rn I can't really thing of a "prettier" solution, because there are certainly tests here that need to run.

Comment (rafaeldtinoco) [reply Makefile:627]: I don't want to block the change, just change it to make more sense. What if we change test-types for something more generic. Check "test-tracee-ebpf"... it contains:

```
./cmd/tracee-ebpf/...
./pkg/...
```

@danielpacak had suggested we included ./pkg/ tests in tracee-ebpf for now and later organized the pkg tests into a single "test" target. @danielpacak WDYT here ?

Comment (rafaeldtinoco) [reply Makefile:627]: BTW, isn't this already being tested by that target ? with "./pkg/..." ?

Comment (NDStrahilevitz) [reply Makefile:627]: > BTW, isn't this already being tested by that target ? with "./pkg/..." ?

No since this is in a different module.
I guess i'd slightly lean on this being in tracee-rules tests, but since `trace` is relevant to tracee-ebpf that doesn't make too much sense either. I can't really see this not being tested seperately for now.

Maybe we need to split tests to ebpf cmd rules cmd and pkg and we can add this under pkg? But that's a larger scope than this PR.

Comment (rafaeldtinoco) [reply Makefile:627]: yes, let's do the following: to add the test into an existing target (tracee-ebpf or tracee-rules) and open an issue to either:

1. create a single test target containing all needed tests
2. or to create multiple targets to each pkg (with a broader one executing them all)

WDYT ?

Comment (rafaeldtinoco) [reply Makefile:627]: (then you can get rid of the workflow changes as well)

Comment (NDStrahilevitz) [reply Makefile:627]: 1. That exists in `make test` which just composes all the other test targets.

I think we just missed restructuring the tests in terms of the new latest structure. Again that would be outside this scope. In terms of the current makefile structure I think how it's done here makes the most sense (at least short term). A PR fixing the entire structure could use the changes done here and reimplement them but I fear if we don't merge this now it will be missed along the way later on.

If we have to include this in one of the other test targets... I guess it would be tracee rules, but I think it should be kept in it's own target.

Comment (rafaeldtinoco) [reply Makefile:627]: Okay, we've synced on Teams about this. We agreed you would include the test in `test-tracee-rules` target and I'm going to open an issue to reformulate the tests targets with the new project structure (since `test-tracee-ebpf` and `test-tracee-rules` are not reflecting the project structure we have today).

Comment (rafaeldtinoco) [reply Makefile]: https://github.com/aquasecurity/tracee/issues/1522

**Pull request merged**
